edisgo/tools/tools.py

```diff
@@ -193,7 +193,109 @@ def drop_duplicated_columns(df, keep="last"):
     return df.loc[:, ~df.columns.duplicated(keep=keep)]
 
 
-def select_cable(edisgo_obj, level, apparent_power):
+def calculate_voltage_diff_per_line(
+    s_max: float | np.ndarray,
+    r_total: float | np.ndarray,
+    x_total: float | np.ndarray,
+    v_nom: float | np.ndarray,
+    sign: int = -1,
+    cos_phi: float = 0.95,
+) -> float | np.ndarray:
+    """
+    Calculate the voltage difference across a line in kV.
+
+    Parameters
+    ----------
+    s_max : float or array-like
+        Apparent power the cable must carry in MVA.
+    r_total : float or array-like
+        Total resistance in Ohms.
+    x_total : float or array-like
+        Total reactance in Ohms.
+    v_nom : float or array-like
+        Nominal voltage in kV.
+    sign : int, optional
+        Sign of the reactance. -1 for inductive and +1 for capacitive. Default is -1.
+    cos_phi : float, optional
+        Power factor (cosine of the phase angle) of the load or generator.
+        Default is 0.95.
+
+    Returns
+    -------
+    float or array-like
+        Voltage difference in kV.
+    """
+    sin_phi = np.sqrt(1 - cos_phi**2)
+    # Calculate the voltage difference using the formula from VDE-AR-N 4105
+    voltage_diff = np.abs(
+        (s_max * 1e6 / (v_nom * 1e3)) * (r_total * cos_phi + sign * x_total * sin_phi)
```

Comment:
Please avoid multiplying and dividing unnecessarily. 1e6/1e3/1e3 all cancels out, so it is not necessary here.

Reply:
Also, please don't return the absolute value here because it would be better to test, whether the voltage rises or drops.